### PR TITLE
Randomized Rumor Spreading initialization test commit

### PR DESCRIPTION
**Pull request by Jazzz42**
## Description
Initial test for Randomized Rumor Spreading

## Status
Test cases, which won't be covered in system test, are going to be tested on unit level using this test suite.

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: tests/RumorSpreading/Test_RumorSpreading.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  }
+  // Copy constructor
+  RRS::RumorHolder dummy_holder(dummy_peerIdSet, 2);
+  RRS::RumorHolder dummy_holder_copy(dummy_holder);
+  /// Rumor Holder == operator check
+  BOOST_CHECK_EQUAL(dummy_holder == dummy_holder_copy, true);
+
+  /// NetworkConfig
+  RRS::NetworkConfig dummy_networkconfig(16);
+  BOOST_TEST_MESSAGE("Rumor Spreading max rounds total: "
```
Comment (jendis) [inline tests/RumorSpreading/Test_RumorSpreading.cpp]: Can't be checked by BOOST_CHECK_EQUAL?
Anyway I would prefer being silent when everything passes.

File: tests/RumorSpreading/Test_RumorSpreading.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                     << dummy_networkconfig.maxRoundsTotal());
+  BOOST_TEST_MESSAGE("Rumor Spreading max rounds in B phase: "
+                     << dummy_networkconfig.maxRoundsInB());
+  BOOST_TEST_MESSAGE("Rumor Spreading max rounds in C phase: "
```
Comment (jendis) [inline tests/RumorSpreading/Test_RumorSpreading.cpp]: Can't be checked by BOOST_CHECK_EQUAL?
Anyway I would prefer being silent when everything passes.

File: tests/RumorSpreading/Test_RumorSpreading.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                     << dummy_networkconfig.maxRoundsInC());
+
+  // Messaging
+  RRS::Message dummy_message_undefined(RRS::Message::Type::UNDEFINED, 0, 0);
+  RRS::Message dummy_message_push(RRS::Message::Type::PUSH, 0, 0);
+  if (dummy_message_push == dummy_message_undefined) {
+    BOOST_TEST_MESSAGE("Dummy undefined message: " << dummy_message_undefined);
+  }
+  if (dummy_message_push != dummy_message_undefined) {
+    BOOST_TEST_MESSAGE("Dummy push message: " << dummy_message_push);
```
Comment (jendis) [inline tests/RumorSpreading/Test_RumorSpreading.cpp]: Can't be checked by BOOST_CHECK_EQUAL?
Anyway I would prefer being silent when everything passes.

Comment (Jazzz42) [reply tests/RumorSpreading/Test_RumorSpreading.cpp]: Done

File: tests/RumorSpreading/Test_RumorSpreading.cpp (hunk 2 above)
Comment (Jazzz42) [reply tests/RumorSpreading/Test_RumorSpreading.cpp]: Done

File: tests/RumorSpreading/Test_RumorSpreading.cpp [hunk 4]
```diff
@@ -65,12 +65,9 @@ BOOST_AUTO_TEST_CASE(RRS_Initialization) {
...
   RRS::Message dummy_message_push(RRS::Message::Type::PUSH, 0, 0);
-  //Test comparison and stream oeprator
-  if (dummy_message_push == dummy_message_undefined) {
-    BOOST_TEST_MESSAGE("RRS Message comparison failed for undefined message: " << dummy_message_undefined);
-  }
-  if (dummy_message_push != dummy_message_push) {
-    BOOST_TEST_MESSAGE("RRS Message comparison failed for push message: " << dummy_message_push);
-  }
+  // Test comparison and stream oeprator
+  BOOST_CHECK_EQUAL(dummy_message_push != dummy_message_undefined, true);
```
Comment (ansnunez) [inline tests/RumorSpreading/Test_RumorSpreading.cpp]: For me, the simplest form `BOOST_CHECK(dummy_message_push != dummy_message_undefined)` seems easier to manage.

Comment (Jazzz42) [reply tests/RumorSpreading/Test_RumorSpreading.cpp]: Done

**Pull request merged**